docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,876 @@
+DPU0       1     Offline              dpu_midplane_link_state        down            Wed 20 Oct 2023 06:52:28 PM UTC
+                                      dpu_booted_state               down            Wed 20 Oct 2023 06:52:28 PM UTC
+                                      dpu_control_plane_state        down            Wed 20 Oct 2023 06:52:28 PM UTC
+                                      dpu_data_plane_state           down            Wed 20 Oct 2023 06:52:28 PM UTC    
+                                      
+root@sonic:~#show system-health DPU 0
+ 
+Name       ID    Oper-Status          State-Detail                   State-Value     Time                               Reason
+DPU0       1     Partial Online       dpu_midplane_link_state        up              Wed 20 Oct 2023 06:52:28 PM UTC
+                                      dpu_booted_state               up              Wed 20 Oct 2023 06:52:28 PM UTC
+                                      dpu_control_plane_state        up              Wed 20 Oct 2023 06:52:28 PM UTC
+                                      dpu_data_plane_state           down            Wed 20 Oct 2023 06:52:28 PM UTC    Pipeline failure
+
+
+```
+#### Pass/Fail Criteria
+ * Verify the following criteria for Pass/Fail:
+ * Online : All states are up
+ * Offline: dpu_midplane_link_state or dpu_booted_state is down
+ * Partial Online: dpu_midplane_link_state is up and dpu_booted_state is up and dpu_control_plane_state is up and dpu_data_plane_state is down
+ * Verify powering down dpu and check for status and powering up again to check the status to show online. 
+
+
+### 1.10 Check the Health of DPUs
+
+#### Steps
+ *  Use command `show system-health detail <DPU_SLOT_NUMBER>` to check the health of the dpu.
+ 
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+
+root@sonic:/home/cisco# show system-health detail <DPU_SLOT_NUMBER>
+
+Device: DPU0
+
+System status summary
+
+  System status LED  green
+  Services:
+    Status: Not OK
+    Not Running: container_checker, lldp
+  Hardware:
+    Status: OK
+
+system services and devices monitor list
+
+Name                       Status    Type
+-------------------------  --------  ----------
+mtvr-r740-04-bf3-sonic-01  OK        System
+rsyslog                    OK        Process
+
+```
+#### Pass/Fail Criteria 
+ * Verify System Status - Green, Service Status - OK, Hardware Status - OK
+ * Stop any docker in DPU and check for Service Status - Not OK and that docker as Not running
+ * Start the docker again and Verify System Status - Green, Service Status - OK, Hardware Status - OK
+ 
+### 1.11 Check reboot cause history
+
+#### Steps
+ *  The "show reboot-cause" CLI on the switch shows the most recent rebooted device, time and the cause. 
+ *  The "show reboot-cause history" CLI on the switch shows the history of the Switch and all DPUs
+ *  The "show reboot-cause history module-name" CLI on the switch shows the history of the specified module
```

Comment:
Same as the above, for the specified mdule, what is the max item for the reboot-cause history?

Reply:
Same as mentioned above and we are considering only for DPU module alone.